WordPress/Classes/Services/PostCoordinator.swift

```diff
@@ -248,19 +248,43 @@ class PostCoordinator: NSObject {
 
 extension PostCoordinator: Uploader {
     func resume() {
-        let service = PostService(managedObjectContext: mainContext)
-
-        service.getFailedPosts { [weak self] posts in
+        let fetcher = FailedPostsFetcher(mainContext)
```

Comment:
I extracted the logic into a class so it can be tested easily. See 60dc5a721173eba04885031d178eb8c8bcd141c8.